olp-cpp-sdk-dataservice-read/src/PrefetchPartitionsHelper.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "PrefetchPartitionsHelper.h"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <olp/core/client/CancellationContext.h>
+#include <olp/core/client/PendingRequests.h>
+#include <olp/core/logging/Log.h>
+#include <olp/dataservice/read/Types.h>
+#include "Common.h"
+#include "DownloadItemsJob.h"
+#include "ExtendedApiResponse.h"
+#include "ExtendedApiResponseHelpers.h"
+#include "QueryPartitionsJob.h"
+#include "TaskSink.h"
+#include "repositories/PartitionsRepository.h"
+
+namespace olp {
+namespace dataservice {
+namespace read {
+namespace {
+const size_t kQueryPartitionsMaxSize = 100u;
+}
+
+void PrefetchPartitionsHelper::Prefetch(
+    std::shared_ptr<DownloadJob> download_job,
+    std::vector<std::string> partitions, QueryFunc query, TaskSink& task_sink,
+    uint32_t priority, client::CancellationContext execution_context) {
+  auto query_job = std::make_shared<QueryPartitionsJob>(
+      std::move(query), nullptr, download_job, task_sink, execution_context,
+      priority);
+
+  size_t query_size = partitions.size() / kQueryPartitionsMaxSize;
+  query_size += (partitions.size() % kQueryPartitionsMaxSize > 0) ? 1 : 0;
```

Comment:
From the previous Veracode issues, I assume that this tool operates strictly on the variable type instead of following the data flow. But the previous attempts were about restricting through conditional statements which are harder to follow in terms of static analysis

Reply:
The previous implementation used a function argument instead of a constant, Veracode is not smart enough to catch that.